src/libstd/sys/windows/process.rs

```diff
@@ -43,13 +43,25 @@ fn mk_key(s: &OsStr) -> OsString {
     })
 }
 
+fn ensure_no_nuls<T: AsRef<OsStr>>(str: T) -> io::Result<T> {
+    let has_nul = {
+        let bytes = str.as_ref().as_inner().inner.as_inner();
```

Comment:
The environment block is built up using `.extend()`, and I wanted to avoid the extra allocation and copy of the keys and values that would be necessary to check the `.encode_wide()` output. See https://github.com/rust-lang/rust/pull/31056/files#diff-9a0a769432651d9c59644e0a8c7f887eR353

We can avoid the repeated allocation by reusing a buffer for the conversion, but we'd still pay the extra copy cost. I'm happy to do this however you prefer, though.